src/main/java/pixelperfect/Spaceship.java

```diff
@@ -73,12 +75,24 @@ public boolean isDead() {
   }
 
   /**
+   * Checks if the ship has reached the end of the Route.
+   * 
+   * @return A Boolean pointing out Victory.
+   */
+  public boolean isVictorious() {
+    return victorious;
+  }
+
+  /**
    * Update the Spaceship and related model classes for the step in the game loop.
    * 
    * @param tpf
    *          Time per frame.
    */
   public void update(float tpf) {
+    if (route.isCompleted(timer)) {
+      this.victorious = true;
+    }
```

Comment:
my thoughts were if you check it in this order, if by time you would actually be done, you would not have a won till the next frame/update where. Basically the timer would be one frame behind. Now as we will have 60+ fps (i guess), it doesnt matter that much, but this order seemed off to me